modules/@apostrophecms/modal/ui/apos/mixins/AposModalTabsMixin.js

```diff
@@ -60,14 +63,16 @@ export default {
 
   watch: {
     tabs() {
-      if ((!this.currentTab) || (!this.tabs.find(tab => tab.name === this.currentTab))) {
-        this.currentTab = this.tabs[0] && this.tabs[0].name;
+      if ((!this.currentTab) || (!this.tabs.find(tab => tab.isVisible === true && tab.name === this.currentTab))) {
+        const firstVisibleTab = this.tabs.find(tab => tab.isVisible === true);
+        this.currentTab = firstVisibleTab && firstVisibleTab.name;
       }
     }
   },
 
   mounted() {
-    this.currentTab = this.tabs[0] ? this.tabs[0].name : null;
+    const firstVisibleTab = this.tabs.find(tab => tab.isVisible === true);
+    this.currentTab = (firstVisibleTab && firstVisibleTab.name) || null;
```

Comment:
This code is redudant, you could move it in a method I guess, like `this.setCurrentTab()`?